substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -863,8 +873,10 @@ impl<T: Config> Module<T> {
             * cu;
         debug::info!("cu cost: {:?}", cu_cost);
 
-        // save total
-        let mut total_cost = su_cost + cu_cost;
+        let mut total_cost = U64F64::from_num(0);
+        if bill_resources {
+            total_cost = su_cost + cu_cost;
+        }
```

Comment:
This can be improved as follows:
- start function by declaring total_cost
- then put the if check around the entirety of the calculation and the addition of the result tto total cost

That way we avoid doing the calculation for CU and SU entirely if we already know the result won't be used anyway. Since the if check needs to happen anyway, this is a serious compute time optimization for that case